File: angle-chan-server/src/main/java/com/ben/dektor/store/ThumbnailStore.java
package com.ben.dektor.store;

import com.ben.dektor.entities.MediaInfo;
import com.ben.dektor.entities.Thumbnail;
import org.springframework.content.commons.repository.ContentStore;
import org.springframework.content.rest.StoreRestResource;
import org.springframework.stereotype.Service;

//@Service
@StoreRestResource(path = "thumbnail")
public interface ThumbnailStore extends ContentStore<Thumbnail, String> {
}

File: angle-chan-server/src/main/kotlin/com/ben/dektor/store/ThumbnailStore.kt
package com.ben.dektor.store

import com.ben.dektor.entities.Thumbnail
import org.springframework.content.commons.repository.ContentStore
import org.springframework.content.rest.StoreRestResource
import org.springframework.stereotype.Component

//@Service
@StoreRestResource(path = "thumbnail")
@Component
interface ThumbnailStore : ContentStore<Thumbnail?, String?>